### PR TITLE
First draft of readme

### PR DESCRIPTION
**Pull request by amirshane**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,51 @@
...
+Protein database search tools such as BLAST are instrumental for research in the life sciences. However, they are slow and are based on surface-level sequence similarity. We are exploring using neural networks to improve the speed and accuracy of finding relevant sequences from these databases. More specifically, we are aiming to learn fixed-length protein embeddings using contextual lenses (https://arxiv.org/pdf/2002.08866.pdf). Generally speaking, a sequence level protein representation, such as a one-hot encoding, is an array of the the form (sequence_length, n) where n is the amino acid embedding dimension. A contextual lens is a (learnable) map from the (sequence_length, n)-array to an (m,)-vector where m is independent of sequence_length. Embeddings are constructed using an encoder function followed by a contextual lens. To learn these embeddings a downstream prediction task is performed using a single feedforward layer. Gradients are backpropagated through all 3 components of the architecture (encoder, lens, predictor) using the Adam optimizer with variable (potentially zero) learning rates per component.
```
Comment (davidBelanger) [inline README.md:5]: start a new paragraph at 'More specifically'

Comment (davidBelanger) [inline README.md:5]: replace 'feedforward' with 'dense'

File: README.md [hunk 2]
```diff
@@ -2,7 +2,51 @@
...
+
+### Encoders
+- One-hot: non-learnable
+- CNN: learnable
+- Transformer: learnable and pretrainable
+
+### Lenses
+- Mean/Max-Pool: non-learnable
+- Linear-Mean/Max-Pool: learnable
```
Comment (davidBelanger) [inline README.md:14]: add links to the corresponding implementations

File: README.md [hunk 3]
```diff
@@ -2,7 +2,51 @@
...
+- GatedConvolution: learnable and self-attentive
+
+## Downstream Task
```
Comment (davidBelanger) [inline README.md:17]: what about the TAPE stuff? Should you mention that?

File: README.md [hunk 4]
```diff
@@ -2,7 +2,51 @@
...
+
+### Pretraining
+We also measure the effect that pretraining has on the performance of a language model encoder. There has been a great deal of interest in measuring the degree to which pretraining protein language models improves their performance on downstream tasks (https://arxiv.org/pdf/1906.08230.pdf). Our results indicate that pretraining offers a substantial boost in performance on the family classification task.
```
Comment (davidBelanger) [inline README.md:21]: Add a reference to the protein_lm repo.

Comment (davidBelanger) [inline README.md:21]: In general, provide more detail about what sort of pretrained models you support and links to relevant related work (ProGen + the FB paper)

File: README.md [hunk 5]
```diff
@@ -2,7 +2,51 @@
...
+
+## Quickstart
```
Comment (davidBelanger) [inline README.md:23]: I'd have a separate quickstart section for how to use the models you've trained. Point to your colab that shows how to load a model and get embeddings from it.

File: README.md (hunk 3 above)
Comment (amirshane) [reply README.md:17]: good question, the TAPE stuff is done in an entirely separate colab notebook, I can add that here and refer to it?

File: README.md (hunk 4 above)
Comment (amirshane) [reply README.md:21]: done, also need to add reference to trembl

File: README.md (hunk 5 above)
Comment (amirshane) [reply README.md:25]: definitely, it's probably best to add this once we have a public bucket going

File: contextual_lenses/blast_baseline.py [hunk 6]
```diff
@@ -2,8 +2,8 @@
 
 Example usage:
 blast_baseline.py \
---train_file=/tmp/formatted_knn_data/5-samples_train_knn_data_families_15001-16000.csv \
---test_file=/tmp/formatted_knn_data/test_knn_data_families_15001-16000.csv
+--train_file=/resources/knn_data/5-samples_train_knn_data_families_15001-16000.csv \
```
Comment (davidBelanger) [inline contextual_lenses/blast_baseline.py:5]: it should be ./resources/...

**Pull request merged**